File: Examples/Excel/VB/Examples/ExcelExamples/Examples/Example06.vb
Imports ExampleBase
Imports LateBindingApi.Core
Imports Excel = NetOffice.ExcelApi
Imports NetOffice.ExcelApi.Enums
Imports NetOffice.VBIDEApi.Enums

Public Class Example06
    Implements IExample

    Dim _hostApplication As ExampleBase.IHost

#Region "IExample Member"

    Public Sub RunExample() Implements ExampleBase.IExample.RunExample

        ' its an example with an own visual control
        ' checkout buttonStartExample_Click

    End Sub

    Public ReadOnly Property Caption As String Implements ExampleBase.IExample.Caption
        Get
            Return IIf(_hostApplication.LCID = 1033, "Example06", "Beispiel06")
        End Get
    End Property

    Public ReadOnly Property Description As String Implements ExampleBase.IExample.Description
        Get
            Return IIf(_hostApplication.LCID = 1033, "Dialogs in Excel", "Dialoge in Excel")
        End Get
    End Property

    Public Sub Connect(ByVal hostApplication As ExampleBase.IHost) Implements ExampleBase.IExample.Connect

        _hostApplication = hostApplication

    End Sub

    Public ReadOnly Property Panel As System.Windows.Forms.UserControl Implements ExampleBase.IExample.Panel
        Get
            Return Me
        End Get
    End Property

#End Region

    Private Sub buttonStartExample_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles buttonStartExample.Click

        ' Initialize NetOffice
        LateBindingApi.Core.Factory.Initialize()

        ' start excel and turn off msg boxes
        Dim excelApplication As New Excel.Application()
        excelApplication.DisplayAlerts = False
        excelApplication.Visible = True

        ' add a new workbook
        Dim workBook As Excel.Workbook = excelApplication.Workbooks.Add()
        Dim workSheet As Excel.Worksheet = workBook.Worksheets(1)

        'show selected window and display user clicks ok or cancel
        Dim returnValue As Boolean
        Dim radioSelectButton As RadioButton = GetSelectedRadioButton()

        Select Case radioSelectButton.Text

            Case "xlDialogAddinManager"

                returnValue = excelApplication.Dialogs(XlBuiltInDialog.xlDialogAddinManager).Show()

            Case "xlDialogFont"

                returnValue = excelApplication.Dialogs(XlBuiltInDialog.xlDialogFont).Show()

            Case "xlDialogEditColor"

                returnValue = excelApplication.Dialogs(XlBuiltInDialog.xlDialogEditColor).Show()

            Case "xlDialogGallery3dBar"

                returnValue = excelApplication.Dialogs(XlBuiltInDialog.xlDialogGallery3dBar).Show()

            Case "xlDialogSearch"

                returnValue = excelApplication.Dialogs(XlBuiltInDialog.xlDialogSearch).Show()

            Case "xlDialogPrinterSetup"

                returnValue = excelApplication.Dialogs(XlBuiltInDialog.xlDialogPrinterSetup).Show()

            Case "xlDialogFormatNumber"

                returnValue = excelApplication.Dialogs(XlBuiltInDialog.xlDialogFormatNumber).Show()

            Case "xlDialogApplyStyle"

                returnValue = excelApplication.Dialogs(XlBuiltInDialog.xlDialogApplyStyle).Show()

        End Select

        ' close excel and dispose reference
        excelApplication.Quit()
        excelApplication.Dispose()

        Dim message As String = String.Format("The dialog returns {0}.", returnValue)
        MessageBox.Show(Me, message, Me.Text, MessageBoxButtons.OK, MessageBoxIcon.Information)

    End Sub

#Region "Helper"

    Private Function GetSelectedRadioButton() As RadioButton

        Dim itemControl As Control
        For Each itemControl In panelSelection.Controls

            If (TypeName(itemControl) = "RadioButton") Then

                Dim radioSelectButton As RadioButton = itemControl
                If (radioSelectButton.Checked) Then
                    Return radioSelectButton
                End If
            End If

        Next itemControl

        Throw (New Exception("No Dialog selected."))

    End Function

#End Region

End Class

File: Examples/Excel/C#/Examples/ExcelExamples/Examples/Example06.cs
using System;
using System.Collections.Generic;
using System.Drawing;
using System.Windows.Forms;
using System.Reflection;
using System.Text;
using ExampleBase;

using NetOffice;
using Excel = NetOffice.ExcelApi;
using NetOffice.ExcelApi.Enums;
using NetOffice.VBIDEApi.Enums;

namespace ExcelExamplesCS4
{
    partial class Example06 : UserControl , IExample
    {
        IHost _hostApplication;

        public Example06()
        {
            InitializeComponent();
        }

        #region IExample Member

        public void RunExample()
        {
            // its an example with an own visual control
            // checkout buttonStartExample_Click
        }

        public void Connect(IHost hostApplication)
        {
            _hostApplication = hostApplication;
        }

        public string Caption
        {
            get { return _hostApplication.LCID == 1033 ? "Example06" : "Beispiel06"; }
        }

        public string Description
        {
            get { return _hostApplication.LCID == 1033 ? "Dialogs in Excel" : "Dialoge in Excel"; }
        }

        public UserControl Panel
        {
            get { return this; }
        }
    
        #endregion

        #region UI Trigger

        private void buttonStartExample_Click(object sender, EventArgs e)
        {           
            // start excel and turn off msg boxes
            Excel.Application excelApplication = new Excel.Application();
            excelApplication.DisplayAlerts = false;
            excelApplication.Visible = true;

            // add a new workbook
            Excel.Workbook workBook = excelApplication.Workbooks.Add();
            Excel.Worksheet workSheet = (Excel.Worksheet)workBook.Worksheets[1];

            // show selected window and display user clicks ok or cancel
            bool returnValue = false;
            RadioButton radioSelectButton = GetSelectedRadioButton();
            switch (radioSelectButton.Text)
            {
                case "xlDialogAddinManager":

                    returnValue = excelApplication.Dialogs[XlBuiltInDialog.xlDialogAddinManager].Show();
                    break;

                case "xlDialogFont":

                    returnValue = excelApplication.Dialogs[XlBuiltInDialog.xlDialogFont].Show();
                    break;

                case "xlDialogEditColor":

                    returnValue = excelApplication.Dialogs[XlBuiltInDialog.xlDialogEditColor].Show();
                    break;

                case "xlDialogGallery3dBar":

                    returnValue = excelApplication.Dialogs[XlBuiltInDialog.xlDialogGallery3dBar].Show();
                    break;

                case "xlDialogSearch":

                    returnValue = excelApplication.Dialogs[XlBuiltInDialog.xlDialogSearch].Show();
                    break;

                case "xlDialogPrinterSetup":

                    returnValue = excelApplication.Dialogs[XlBuiltInDialog.xlDialogPrinterSetup].Show();
                    break;

                case "xlDialogFormatNumber":

                    returnValue = excelApplication.Dialogs[XlBuiltInDialog.xlDialogFormatNumber].Show();
                    break;

                case "xlDialogApplyStyle":

                    returnValue = excelApplication.Dialogs[XlBuiltInDialog.xlDialogApplyStyle].Show();
                    break;

                default:
                    throw (new Exception("Unkown dialog selected."));

            }

            string message = string.Format("The dialog returns {0}.", returnValue);
            MessageBox.Show(this, message, this.Text, MessageBoxButtons.OK, MessageBoxIcon.Information);

            // close excel and dispose reference
            excelApplication.Quit();
            excelApplication.Dispose();
        }
        
        #endregion

        #region Helper

        private RadioButton GetSelectedRadioButton()
        {
            foreach (Control itemControl in panelSelection.Controls)
            {
                RadioButton radioSelectButton = itemControl as RadioButton;
                if (null != radioSelectButton)
                {
                    if (radioSelectButton.Checked)
                        return radioSelectButton;
                }
            }

            throw (new InvalidOperationException("No Dialog selected."));
        }

        #endregion
    }
}
